__tests__/app.test.js

```diff
@@ -213,3 +213,57 @@ describe('GET /api/articles/:article_id/comments', () => {
       })
   })
 })
+
+describe('POST /api/articles/:article_id/comments', () => {
+  test('status:201, responds with the object of posted comment', () => {
+    const commentData = {
+      username: 'rogersop',
+      body: 'British following properties',
+    }
+    return request(app)
+      .post('/api/articles/3/comments')
+      .send(commentData)
+      .expect(201)
+      .then(({ body }) => {
+        expect(body).toMatchObject({
+          comment: {
+            comment_id: 19,
+            article_id: 3,
+            votes: 0,
+            created_at: expect.any(String),
+            author: 'rogersop',
+            body: 'British following properties',
+          },
+        })
+      })
+  })
+  test('Status:400, when invalid article_id is passed', () => {
+    const commentData = {
+      username: 'rogersop',
+      body: 'British following properties',
+    }
+    return request(app)
+      .post('/api/articles/333/comments')
```

Comment:
would this be a 400 - bad request or a 404 - not found? The id is correct it just doesnt exist yet!

Reply:
I reckon 404 would be more appropriate

Reply:
Actually I think 400 is absolutely fine. We are creating a new comment at this endpoint, and  we can't create a comment with invalid  article_id. So basically we are providing bad data. Is that right?